rpc_new/util.ts

```diff
@@ -0,0 +1,12 @@
+// TODO: expose these utils from `util/mod.ts`
+
+export type Listener<Event, This = any> = (this: This, event: Event) => void;
+
+export function contramapListener<From, Into>(
```

Comment:
Should probably pass `this` through

Reply:
True... come to think of it, I'm not sure if we'll end up using this util. `Listener` is much simpler now (the mapping requires little code) and we'll likely want to customize the context depending on the effect (aka. `apply({ /* customizing this */ }, [...])`). Still, will pass it through for now.